### ISSUE TITLE
[feature request] Show the type of syntax highlighting that is performed

### ISSUE THREAD

**Issue by yurivict** [gogits/gogs#2207]
Currently it is sometimes hard to tell what syntax is this and what highlighting is performed.

**Comment by Unknwon** [comment]
Thanks your feedback!

What is the matter of knowing which syntax is applied?

PS, you can see it from web console:

![image](https://cloud.githubusercontent.com/assets/2946214/11828915/a397aa88-a366-11e5-9474-0b76b996d878.png)

**Comment by yurivict** [comment]
It is good to let the user know what highlighting is used, because this isn't always obvious. In small text in the corner is okay.

**Comment by zacheryph** [comment]
Can you give an example of where this is not obvious? Or where its highlighting as a non-proper language?  Is the file extension not an indicator of what the file is being highlighted as?

**Comment by yurivict** [comment]
Most people don't know what most extensions mean. Also some extensions overlap. There are also situations when texts with different syntaxes are embedded in each other, and for example phabricator is able to highlight syntax properly in such situation.

This is why it is a good idea to show the syntax name.

**Comment by andreynering** [comment]
@zacheryph If you are going to try it, this function should do the trick:

https://github.com/gogits/gogs/blob/master/models/git_diff.go#L170-L172

> There are also situations when texts with different syntaxes are embedded in each other, and for example phabricator is able to highlight syntax properly in such situation.

Gogs uses Highlight.js and I think this already works.

**Comment by spectre013** [comment]
Added this and the line count in the header. it bases it off the highlight type which is the primary source of the highlighting so if you have an HTML document with JavaScript in it it will still say HTML. 

Image of the changes attached if are okay with them I will issue a pull request. I am not sold with the word "highlighting"  in there so open to suggestions there. 

![gog-fileheaderupdate](https://cloud.githubusercontent.com/assets/1624093/23964133/13e08a7a-0979-11e7-9e4e-2f4d53acd4b2.PNG)

**Comment by Unknwon** [comment]
@spectre013 Yes, please send a PR. The functionality in screenshot looks good. More minor details can be fixed/discussed during the PR.

**Comment by jxsl13** [comment]
Syntax Highlighting in java is quite bad :c 
![33fcsqf](https://cloud.githubusercontent.com/assets/371944/24818106/6cbcc9a6-1bdf-11e7-8863-c84b2b887997.png)
would be cool to have Microsofts Monaco editor as suggested in another issue.

**Comment by Unknwon** [comment]
@jxsl13 you should send issue to highlightJS and the comment is irrelevant to this issue.

**Comment by jxsl13** [comment]
gonna delete this afterwards, but will the highlight engine be changed in the near future :)?

**Comment by Unknwon** [comment]
@jxsl13 use Monaco doesn't change the fact highlight for code file is using highlightJS.

**Comment by jxsl13** [comment]
are there any alternatives?

**Comment by Unknwon** [comment]
@jxsl13 not in any time soon and not planed. I'm going to delete our comments because we're spamming people who are watching for this thread. Discussion goes to user forum.
